tests/Aspire.Hosting.Tests/Dashboard/ResourcePublisherTests.cs

```diff
@@ -131,6 +131,35 @@ public async Task DeletesRemoveFromSnapshot()
         await cts.CancelAsync();
     }
 
+    [Fact]
+    public async Task CancelledSubscriptionIsCleanedUp()
```

Comment:
Can you replicate the other two tests from https://github.com/dotnet/aspire/pull/1576/files#diff-b609d38a456686a2ab0a1b724803980a09de0151b8dc3e0e11408ee43755b076?

A test that subscriptions are cleaned up when the publisher is cleaned up (the constructor CT is raised) is important.

Reply:
I've added `DashboardClientTests`.